packages/init/index.ts

```diff
@@ -21,3 +23,38 @@ export default function initializeInquirer(...args: string[]): Function | void {
 	}
 	return npmPackagesExists(packages);
 }
+
+/**
+ * UI Manager function which takes answers and dependendencies as input
+ * and generates required scaffolds
+ * @param dependencies - array of packages needed to be installed
+ * @param config - Config object to be returned from UI
+ */
+
+export function UIManager(dependencies: string[], config: ITransformConfig) {
+	const npm = require("npm");
+	const returnObject: {
+		errors: String[],
+	} = {errors: []};
+
+	// Install Dependencies
+	npm.load((err) => {
```

Comment:
It was under the yeoman generator which I think can't be reused here.
Any other way to achieve this then? 
@ematipico

Reply:
My vision was to use the generator by passing the correct configuration. Let's see what we can do

Reply:
So, at the moment the generator doesn't expose functions to add more configurations. We could create some method to do so, so we don't need the prompting. Also, inside the utils package we have an utility to identify the package manager to use. Anyhow, the best option is to use the generator that we already have, extend it by adding a function that allows us to configurations (plus one to add a configuration to the top scope) and launch the install method of the generator

Reply:
We can't stop execution of prompting() while execution of generator ( as per my knowledge and understanding ).

So use of generator would ultimately force us to use child_process which I think is the problem from which we started to solve here.

Maybe I am wrong here but currently I am not in interest of using generators.

If you know some way to stop execution of prompting() @ematipico then it would make our work to easy and I am happy to make changes accordingly.